pangeo_forge_recipes/transforms.py

```diff
@@ -591,23 +595,22 @@ class WriteCombinedReference(beam.PTransform, ZarrWriterMixin):
     concat_dims: List[str]
     identical_dims: List[str]
     mzz_kwargs: dict = field(default_factory=dict)
+    remote_options: Optional[Dict] = field(default_factory=dict)
+    remote_protocol: Optional[str] = None
     target_root: Union[str, FSSpecTarget, RequiredAtRuntimeDefault] = field(
         default_factory=RequiredAtRuntimeDefault
     )
     output_file_name: str = "reference.json"
 
     def expand(self, references: beam.PCollection) -> beam.PCollection[zarr.storage.FSStore]:
-        # unpack fsspec options that will be used below for transforms without dep injection
-        storage_options = self.target_root.fsspec_kwargs  # type: ignore[union-attr]
-        remote_protocol = self.target_root.get_fsspec_remote_protocol()  # type: ignore[union-attr]
         return (
             references
             | CombineReferences(
                 concat_dims=self.concat_dims,
                 identical_dims=self.identical_dims,
-                target_options=storage_options,
-                remote_options=storage_options,
-                remote_protocol=remote_protocol,
+                target_options=self.remote_options,
```

Comment:
This is incredibly confusing but I'm actually not sure, still, why remote_options and target_options can be the same here. They're passed as distinctly named arguments here: https://github.com/pangeo-forge/pangeo-forge-recipes/blob/main/pangeo_forge_recipes/transforms.py#L447-L448 and in https://fsspec.github.io/kerchunk/reference.html#kerchunk.combine.MultiZarrToZarr it says:

>target_options – dict Storage options for opening path

which makes me think target_options (where the reference files are, say s3://gcorradini-...) should not be the same as remote_options, which doesn't have a docstring in MultiZarrToZarr but I have to assume are credentials for the bucket which contains the data files (e.g. s3://gesdisc-cumulus-protected)

But we saw that this worked so maybe I'm missing something you explained previously.

Reply:
I know 😆. At the `fsspec` level inside `MultiZarrToZarr` it is very confusing where `target_options` vs `remote_options` are used which is why we're covering our butts here by passing all of them through (this is a theme in the recipes code if you poke around enough)

But it does work indeed as this Flink run shows: https://github.com/NASA-IMPACT/veda-pforge-job-runner/actions/runs/8057334009